### PR TITLE
Sdit 1894 orders

### PR DESCRIPTION
**Pull request by lbennett-moj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingService.kt [hunk 1]
```diff
@@ -217,6 +219,7 @@ class SentencingService(
       }
     }
 
+  // TODO not currently updating the EXISTING offendercharge - eg with new offenceresultcode
   @Audit
```
Comment (lbennett-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingService.kt:223]: will need to update the underlying offender charge as well as creating the court event charge

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/CourtEvent.kt [hunk 2]
```diff
@@ -118,7 +118,7 @@ class CourtEvent(
   val directionCode: DirectionType? = null,
 
   @Convert(converter = YesNoConverter::class)
-  val holdFlag: Boolean? = null,
+  val holdFlag: Boolean? = false,
```
Comment (lbennett-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/CourtEvent.kt:121]: replicating pnomis defaults

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingResourceIntTest.kt [hunk 3]
```diff
@@ -1001,6 +1002,7 @@ class SentencingResourceIntTest : IntegrationTestBase() {
           .jsonPath("beginDate").isEqualTo("2023-01-01")
           .jsonPath("createdByUsername").isNotEmpty
           .jsonPath("createdDateTime").isNotEmpty
+          .jsonPath("courtEvents[0].courtOrders[0].id").exists()
```
Comment (lbennett-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingResourceIntTest.kt:1005]: court order created

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingResourceIntTest.kt [hunk 4]
```diff
@@ -961,6 +961,7 @@ class SentencingResourceIntTest : IntegrationTestBase() {
                       offenceDate = LocalDate.of(2023, 1, 2),
                       offenceEndDate = LocalDate.of(2023, 1, 3),
                       offenceCode = "HP03001",
+                      resultCode1 = "1002",
```
Comment (lbennett-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingResourceIntTest.kt:964]: final active result will trigger an order creation

**Review by steverendell — APPROVED** [review]
Lgtm

**Pull request merged**